grafana-plugin/src/components/ExtensionLinkMenu/ExtensionLinkMenu.tsx

```diff
@@ -0,0 +1,131 @@
+import React, { ReactElement, useMemo } from 'react';
+
+// Note: `PluginExtensionLink` is available in Grafana>=10.0.
+// @ts-expect-error
+import { locationUtil, PluginExtensionLink } from '@grafana/data';
+import { Menu } from '@grafana/ui';
+
+import { PluginBridge, SupportedPlugin } from 'components/PluginBridge/PluginBridge';
+
+type Props = {
+  extensions: PluginExtensionLink[];
+  // We require this to be passed in so we can continue to
+  // create a custom Declare incident link. Once the Incident plugin
+  // registers its own extension link, we can remove this.
+  declareIncidentLink?: string;
+  grafanaIncidentId?: string;
+};
+
+export function ExtensionLinkMenu({ extensions, declareIncidentLink, grafanaIncidentId }: Props): ReactElement | null {
+  const { categorised, uncategorised } = useExtensionLinksByCategory(extensions);
+  const showDivider = uncategorised.length > 0 && Object.keys(categorised).length > 0;
+
+  return (
+    <Menu>
+      <>
+        <DeclareIncidentMenuItem
+          extensions={extensions}
+          declareIncidentLink={declareIncidentLink}
+          grafanaIncidentId={grafanaIncidentId}
+        />
+        {Object.keys(categorised).map((category) => (
+          <Menu.Group key={category} label={truncateTitle(category, 25)}>
+            {renderItems(categorised[category])}
+          </Menu.Group>
+        ))}
+        {showDivider && <Menu.Divider key="divider" />}
+        {renderItems(uncategorised)}
+      </>
+    </Menu>
+  );
+}
+
+// This menu item is a temporary workaround for the fact that the Incident plugin doesn't
+// register its own extension link.
+// TODO: remove this once Incident is definitely registering its own extension link.
+function DeclareIncidentMenuItem({ extensions, declareIncidentLink, grafanaIncidentId }: Props): ReactElement | null {
+  const declareIncidentExtensionLink = extensions.find(
+    (extension) => extension.pluginId === 'grafana-incident-app' && extension.title === 'Declare incident'
+  );
+  if (
+    // Don't show a custom Declare incident button if the Grafana Incident plugin already configured one.
+    declareIncidentExtensionLink !== undefined ||
+    // Don't show a custom Declare incident button if there's no valid link.
+    declareIncidentLink === undefined ||
+    // Don't show the button if an incident has already been declared from this alert group.
+    grafanaIncidentId !== null
```

Comment:
What if it is `undefined`?

In general maybe those checks could be a bit less narrowed and could cover all falsy values, does it make sense? For example:
```ts
if (
    // Don't show a custom Declare incident button if the Grafana Incident plugin already configured one.
    declareIncidentExtensionLink  ||
    // Don't show a custom Declare incident button if there's no valid link.
    !declareIncidentLink ||
    // Don't show the button if an incident has already been declared from this alert group.
    grafanaIncidentId
  }
```

Reply:
Yep makes sense. I've pushed a fix using your recommendation 🙂